File: yh_star/src/yh_star_sub.cpp
#include "ros/ros.h"
#include "yh_star/yh_star_msg.h"


void msgCallback(const yh_star::yh_star_msg::ConstPtr& msg)
{
    ROS_INFO("receive msg = %d", msg->stamp.sec);
    ROS_INFO("receive msg = %d", msg->stamp.nsec);
    ROS_INFO("receive msg = %d", msg->data);

    for(int i= 1; i <= msg->data; i++)
    {
        if (msg->data / 2 >= i) 
        {
            for(int j = 0; j < i; j++)
            {
                printf("*");
            }
        }
        else
        {
            for(int j = 0; j < msg->data - i; j++)
            {
                printf("*");
            }
        }
        printf("\n");        
    }
    
}

int main(int argc, char** argv)
{
    ros::init(argc, argv, "yh_star_sub");
    ros::NodeHandle nh;

    ros::Subscriber sub =  nh.subscribe("yh_star_topic", 100, msgCallback);

    ros::spin();  //프로그램 종료 방지 대기

    return 0;

}

File: yh_star/scripts/yh_star_sub.py
#!/usr/bin/python
# -*- coding: utf-8 -*-

import rospy
from yh_star.msg import yh_star_msg

def msgCallback(msg):
    rospy.loginfo("receieve message: %d", msg.stamp.secs)
    rospy.loginfo("receieve message: %d", msg.stamp.nsecs)
    rospy.loginfo("receieve message: %d", msg.data)

    for i in range(1, msg.data):
        if(msg.data / 2 >= i):
            stra = '*' * i
            print(stra)
        else:
            strb = '*' * (msg.data - i)
            print(strb)

    

def listner():
    rospy.init_node("yh_star_sub", anonymous=True)
    rospy.Subscriber("yh_star_topic", yh_star_msg, msgCallback)

    rospy.spin()

if __name__ == "__main__":
    listner()
